### DISCUSSION TITLE
API - Impersonate Users using Dynamic API Key

### DISCUSSION THREAD

**Discussion by fntyler** [drakkan/sftpgo#1318] [Q&A]
How do you impersonate a user using an API Key ("dynamic")?

[docs - rest-api.md](https://github.com/drakkan/sftpgo/blob/main/docs/rest-api.md)

Taking from the latest documentation regarding the REST API, a "dynamic" API Key is one that meets the following criteria:

* Scope = 2
* Has no user's username or admin's username associated with it.

I assume that the association is completed at the time of creating the API Key when supplying a valid BearToken (JWT) to the endpoint: `/apikeys`

```
POST /apikeys

REQUEST BODY
{
   "name": "johntheuser",
   "scope": 2,
   "user": ""
}
```
Scope Option (int): 
* 1 - admin scope. The API key will be used to impersonate an SFTPGo admin
* 2 - user scope. The API key will be used to impersonate an SFTPGo user

User Option (str):
* username associated with this API key. If empty and the scope is "user scope" the key can impersonate any user

Admin Option (str):
* admin associated with this API key. If empty and the scope is "admin scope" the key can impersonate any admin

When I attempt to impersonate the user, I receive the error, `{"message":"API key authentication is not allowed"}` , using the following settings & API Endpoint:

`johntheuser` has the property: `allow_api_key_auth": true`
API Key Format: `API_KEY_ID.API_KEY_SECRET.johntheuser`

```
GET user/profile
X-SFTPGO-API-KEY: API_KEY_ID.API_KEY_SECRET.johntheuser
```

I can successfully impersonate SFTPGo Admins.

**Comment by Mickaeljpn** [comment]
Same issu here.
